src/storage/local_storage/local_rel_table.cpp

```diff
@@ -181,23 +178,45 @@ column_id_t LocalRelTable::rewriteLocalColumnID(RelDataDirection direction, colu
 }
 
 bool LocalRelTable::scan(Transaction* transaction, TableScanState& state) const {
-    const auto& relScanState = state.cast<RelTableScanState>();
+    auto& relScanState = state.cast<RelTableScanState>();
     KU_ASSERT(relScanState.localTableScanState);
     auto& localScanState = *relScanState.localTableScanState;
-    KU_ASSERT(localScanState.rowIndices.size() >= localScanState.nextRowToScan);
-    const auto numToScan = std::min(localScanState.rowIndices.size() - localScanState.nextRowToScan,
-        DEFAULT_VECTOR_CAPACITY);
-    if (numToScan == 0) {
-        return false;
-    }
-    for (auto i = 0u; i < numToScan; i++) {
-        localScanState.rowIdxVector->setValue<row_idx_t>(i,
-            localScanState.rowIndices[localScanState.nextRowToScan + i]);
+    while (true) {
```

Comment:
Looks correct but u are the one knows the details. I'm not gonna review line by line for this file.